lib/datadog/appsec/reactive/operation.rb

```diff
@@ -40,6 +41,16 @@ def finalize
           Thread.current[:datadog_security_active_operation] = parent
         end
 
+        private
+
+        def select_reactive_engine(reactive, parent)
+          return reactive if reactive
+
+          return parent.reactive unless parent.nil?
+
+          Reactive::Engine.new
+        end
```

Comment:
I believe this block is more readable than the previous one: 
`reactive_engine || (parent.reactive unless parent.nil?) || Reactive::Engine.new`